src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -50,255 +50,272 @@ DatabaseShardImp::DatabaseShardImp(
         app.config().section(ConfigSection::shardDatabase()),
         j)
     , app_(app)
+    , parent_(parent)
+    , taskQueue_(std::make_unique<TaskQueue>())
     , earliestShardIndex_(seqToShardIndex(earliestSeq()))
     , avgShardFileSz_(ledgersPerShard_ * kilobytes(192))
 {
 }
 
-
 DatabaseShardImp::~DatabaseShardImp()
 {
     // Stop threads before data members are destroyed
     stopThreads();
 
     // Close backend databases before destroying the context
-    std::lock_guard lock(m_);
-    complete_.clear();
-    if (incomplete_)
-        incomplete_.reset();
-    preShards_.clear();
+    std::lock_guard lock(mutex_);
+    shards_.clear();
     ctx_.reset();
 }
 
 bool
 DatabaseShardImp::init()
 {
     using namespace boost::filesystem;
-
-    std::lock_guard lock(m_);
+    using namespace boost::beast::detail;
     auto fail = [j = j_](std::string const& msg)
     {
         JLOG(j.error()) <<
             "[" << ConfigSection::shardDatabase() << "] " << msg;
         return false;
     };
+    std::lock_guard lock(mutex_);
+
+    {
+        std::lock_guard lock(mutex_);
```

Comment:
Locking the same mutex twice? Also, if this is intended for some reason do we advise name shadowing here?

Reply:
This was a mistake, accidental typo when narrowing the lock scope.